File: ruby/detect_capital_use.rb
# frozen_string_literal: true

# @param {String} word
# @return {Boolean}
def detect_capital_use(word)
  return true if word.match(/\A[A-Z]+\z/)

  return true if word.match(/\A[a-z]+\z/)

  return true if word.match(/\A[A-Z][a-z]+\z/)

  false
end

File: python/detect_capital_use.py
import re

class Solution:
    def detectCapitalUse(self, word: str) -> bool:
        if re.match(r'^[A-Z]+$', word):
            return True
        elif re.match(r'^[a-z]+$', word):
            return True
        elif re.match(r'^[A-Z][a-z]+$', word):
            return True
        else:
            return False
